File: test/fixtures/go-app-merge/merge.go
// Package merge combines two carts, summing quantities for matching item
// ids.
package merge

type Item struct {
	ID  string
	Qty int
}

// MergeCarts has a seeded bug: it concatenates without combining matching
// ids, so a shared item shows up as two separate lines instead of one with
// a summed quantity.
func MergeCarts(cartA, cartB []Item) []Item {
	return append(append([]Item{}, cartA...), cartB...)
}

File: test/fixtures/py-app-merge/merge.py
"""Merges two carts, combining quantities for matching item ids."""


def merge_carts(cart_a, cart_b):
    # Bug: concatenates without combining matching ids, so a shared item shows
    # up as two separate lines instead of one with a summed quantity.
    return [*cart_a, *cart_b]
